docs/logging.md

```diff
@@ -0,0 +1,102 @@
+# Logging in the chs-delta-api
+This application uses the following standard format to log messages using the chs.go logger.
+
+## Notable logs
+The following sub sections will cover all notable logs that are output by this service.
+
+
+#### Validation fails
+if a request fails validation you will get the following log:
+
+```go
+{"context":"context_id","created":"date_time_stamp","data":{"message":"message_text"},"event":"info","namespace":"chs-delta-api"}
+{"context":"context_id","created":"date_time_stamp","data":{"error":{},"message":"ch_errors_object_array"},"event":"error","namespace":"chs-delta-api"}
+```
+
+The following message will be logged:
+```go
+"message":"Request validated. Errors found."
+```
+
+followed by a message containing the errors found.
+```go
+"message":"{Error:value is not one of the allowed values, ErrorValues:map[secure_director:test], Location:officers.0.secure_director, LocationType:json-path, Type:ch:validation},"
+```
+
+
+The following variables are of interest to Kibana:
+
+`"context":"context_id"` - This is the only variable which can be used to track a request through the full end to end journey.
+
+With an example of this being:
+```go
+{"context":"Dy7rFtAq3G5G9m60MZ1jlgkgeyLD","created":"2021-09-14T11:03:02.042891994+01:00","data":{"message":"Request validated. Errors found."},"event":"info","namespace":"chs-delta-api"}
+{"context":"Dy7rFtAq3G5G9m60MZ1jlgkgeyLD","created":"2021-09-14T11:03:02.045197901+01:00","data":{"error":{},"message":"{Error:value is not one of the allowed values, ErrorValues:map[secure_director:test], Location:officers.0.secure_director, LocationType:json-path, Type:ch:validation},"},"event":"error","namespace":"chs-delta-api"}
+```
+---
+
+#### Message sent to Kafka
+```go
+{"context":"context_id","created":"date_time_stamp","data":{"message":"message_text","offset":offset_int,"partition":partition_int,"topic":"topic_choice"},"event":"info","namespace":"chs-delta-api"}
+```
+The following message will be logged:
+```go
+"message":"Sent message"
+```
+
+The following variables are of interest to Kibana:
+
+`"context":"context_id"` - This is the only variable which can be used to track a request through the full end to end journey.
+
+`"offset":offset_int, "partition":partition_int` - These values are only returned if a message has successfully been sent to Kafka.
+
+With an example of this being:
+```go
+{"context":"zWJcenQdHgzRmoK1hL_ridjiybm1","created":"2021-09-13T12:19:29.825429905+01:00","data":{"message":"Sent message","offset":1,"partition":6,"topic":"officers-delta"},"event":"info","namespace":"chs-delta-api"}
+```
+---
+
+#### Message fails to be sent to Kafka
```

Comment:
Could you make the point that the messages vary quite a bit. Perhaps give the two examples shared by yourself & Faisal?

Reply:
Both your first comment and this one have been implemented. Thanks for the review :)